homeassistant/components/nuki/__init__.py

```diff
@@ -1,28 +1,57 @@
 """The nuki component."""
 
+import asyncio
 from datetime import timedelta
+import logging
 
-import voluptuous as vol
+import async_timeout
+from pynuki import NukiBridge
+from pynuki.bridge import InvalidCredentialsException
+from requests.exceptions import RequestException
 
-from homeassistant.components.lock import DOMAIN as LOCK_DOMAIN
-from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant import exceptions
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_REAUTH
 from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN
-import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    DATA_BRIDGE,
+    DATA_COORDINATOR,
+    DATA_LOCKS,
+    DATA_OPENERS,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    ERROR_STATES,
+)
 
-from .const import DEFAULT_PORT, DOMAIN
+_LOGGER = logging.getLogger(__name__)
 
-PLATFORMS = ["lock"]
+PLATFORMS = ["binary_sensor", "lock"]
 UPDATE_INTERVAL = timedelta(seconds=30)
 
-NUKI_SCHEMA = vol.Schema(
-    vol.All(
-        {
-            vol.Required(CONF_HOST): cv.string,
-            vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-            vol.Required(CONF_TOKEN): cv.string,
-        },
-    )
-)
+
+def _get_bridge_devices(bridge):
+    # pylint: disable=protected-access
```

Comment:
this is really bad architecture. Please link an upstream PR which resolve that we don't need that in the next RL

Reply:
The library provides `locks` and `openers` properties that map to these exact function calls, so why aren't those used instead?